### PR TITLE
Base.ProviderToProviderSubType Validation

### PR DESCRIPTION
**Pull request by oajimenez**
Changed select statement to lateral flatten format, most things within margin now with the exception of trivial columns like InsertedOn, InsertedBy, LastUpdateDate which make large margins misleading. 

**Important thing is that total rows matches the total amount of providers (6.5M).**

### CODE REVIEW

**Review by carresemata — CHANGES_REQUESTED** [review]
join in the merge with providertosubtypeid

File: migration_original/ODS1Stage/tables/Base/ProviderToProviderSubType/spu_translated_ProviderToProviderSubType.sql [hunk 1]
```diff
@@ -35,21 +36,31 @@ begin
...
-                            join base.providersubtype as PST on pst.providersubtypecode = json.providersubtype_ProviderSubTypeCode
-                        where
-                            PROVIDER_PROFILE is not null
-                            and ProviderSubType_ProviderSubTypeCode is not null
-                            and ProviderID is not null 
-                        qualify row_number() over( partition by ProviderID, ifnull(ProviderSubType_ProviderSubTypeCode, 'ALT') order by CREATE_DATE desc) = 1 $$;
+                        from cte_providersubtype as json
+                        join base.provider as p on p.providercode = json.providercode
+                        join base.providersubtype as pst on pst.providersubtypecode = json.providersubtype_ProviderSubTypeCode
+                        where json.providersubtype_ProviderSubTypeCode is not null
```
Comment (carresemata) [inline migration_original/ODS1Stage/tables/Base/ProviderToProviderSubType/spu_translated_ProviderToProviderSubType.sql:61]: you can add this where clause in the select above ( in the cte )

**Review by carresemata — CHANGES_REQUESTED** [review]

File: migration_original/ODS1Stage/tables/Base/ProviderToProviderSubType/spu_translated_ProviderToProviderSubType.sql (hunk 1 above)
Comment (oajimenez) [reply migration_original/ODS1Stage/tables/Base/ProviderToProviderSubType/spu_translated_ProviderToProviderSubType.sql:61]: I don't see the benefit of doing this. I ran the select statement a couple of times and there weren't any performance gains, and in terms of organizational purposes it makes more sense to have the CTE have the raw flattened data and the filters below in the final select statement.

**Comment by oajimenez** [comment]
> join in the merge with providertosubtypeid

 ✅

**Review by carresemata — APPROVED** [review]
good, you are almost done!

**Pull request merged**